speech/google/cloud/speech/operation.py

```diff
@@ -14,120 +14,55 @@
 
-        :rtype: bool
-        :returns: True if already completed, else false.
-        """
-        return self._complete
+        This mostly does what the base class does, but all populates
+        results.
 
-    @property
-    def metadata(self):
-        """Metadata of operation.
+        :type operation_pb:
+            :class:`~google.longrunning.operations_pb2.Operation`
+        :param operation_pb: Protobuf to be parsed.
 
-        :rtype: :class:`~google.cloud.speech.metadata.Metadata`
-        :returns: Instance of ``Metadata``.
+        :raises ValueError: If there is more than one entry in ``results``.
         """
-        return self._metadata
+        super(Operation, self)._update_state(operation_pb)
 
-    @property
-    def results(self):
-        """Results dictionary with transcript information.
+        result_type = operation_pb.WhichOneof('result')
+        if result_type != 'response':
+            return
 
-        :rtype: dict
-        :returns: Dictionary with transcript and confidence score.
-        """
-        return self._results
-
-    def poll(self):
-        """Check if the operation has finished.
-
-        :rtype: bool
-        :returns: A boolean indicating if the current operation has completed.
-        :raises: :class:`ValueError <exceptions.ValueError>` if the operation
-                 has already completed.
-        """
-        if self.complete:
-            raise ValueError('The operation has completed.')
+        pb_results = self.response.results
+        if len(pb_results) != 1:
+            raise ValueError('Expected exactly one result, found:',
+                             pb_results)
 
-        path = 'operations/%s' % (self.name,)
-        api_response = self.client.connection.api_request(method='GET',
-                                                          path=path)
-        self._update(api_response)
-        return self.complete
-
-    def _update(self, response):
-        """Update Operation instance with latest data from Speech API.
-
-        .. _speech_operations: https://cloud.google.com/speech/reference/\
-                               rest/v1beta1/operations
-
-        :type response: dict
-        :param response: Response from Speech API Operations endpoint.
-                         See: `speech_operations`_.
-        """
-        metadata = response.get('metadata', None)
-        raw_results = response.get('response', {}).get('results', None)
-        results = []
-        if raw_results:
-            for result in raw_results:
-                for alternative in result['alternatives']:
-                    results.append(Transcript.from_api_repr(alternative))
-        if metadata:
-            self._metadata = Metadata.from_api_repr(metadata)
-
-        self._results = results
-        self._complete = response.get('done', False)
+        result = pb_results[0]
+        self.results = [Transcript.from_pb(alternative)
```

Comment:
Do we really need to use `Transcript` for this now? Since it's going to get renamed to `Alternative` which would pretty much be the same as [google.cloud.speech.v1beta1.SpeechRecognitionAlternative(https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.SpeechRecognitionAlternative)?